src/Umbraco.Web.UI.Client/src/packages/documents/documents/tree/tree-item/document-tree-item.element.ts

```diff
@@ -76,11 +76,14 @@ export class UmbDocumentTreeItemElement extends UmbTreeItemElementBase<UmbDocume
 	}
 
 	override renderIconContainer() {
+		const icon = this.item?.documentType.icon;
+		const iconWithoutColor = icon?.split(' ')[0];
```

Comment:
The `icon?.split(' ')[0];` logic is shared, should it be a method on the base class that can handle the logic?

Reply:
I would prefer if this got moved all the way to the server. If that is not possible, we should map the data when we receive it. Instead of a string with both, we should have name and color in separate props. I think we will need this in more cases.

Reply:
To me it seems like a UI concern that a picked icon and colour has to change, but I can shared your idea with the rest of the team

Reply:
Otherwise you only talk about splitting colour and icon